frontend/test/pytest/test_gradient.py

```diff
@@ -261,6 +261,14 @@ def f3(x: float):
     assert np.allclose(result[0]["helloworld"], expected[0]["helloworld"])
     assert np.allclose(result[1]["helloworld"], expected[1]["helloworld"])
 
+    @qjit
+    def f4(x: float, y: float, z: float):
+        return 100 * x + 200 * y + 300 * z
+
+    result = qjit(value_and_grad(f4))(0.1, 0.2, 0.3)
+    expected = (140, 100)
+    assert np.allclose(result, expected)
```

Comment:
@paul0403 good to also add a test to show that different `argnum` values are working!

Reply:
Unfortunately argnum is not working now (the mlir separate bug you observed in the issue), Romain and I are looking into it rn

Reply:
ah, so it only works with `argnum=0`?

Reply:
It worked only with one argnum because the split between values and gradient is equal. We had to tell MLIR that the equal split assumption is wrong. The fix is coming.

Reply:
fixed now! @josh146

Reply:
that was fast! thank you